### ISSUE TITLE
Login not working, only gives an error page.

### ISSUE THREAD

**Issue by KandiPanda** [twireapp/Twire#155]
Running on Android 4.1.2 on a Casio C811.

Tried with both 2.9.0 and 2.519, (The latter mainly because skipping the broken login in 2.9.0 just crashes it.)

Either version, attempting to log in just takes me to a screen reading:
```
Webpage not available

The webpage at https://www.twitch.tv/login?client_id=(bunch of stuff here) might be temporarily down or it may have moved permanently to a new web address.
```

**Comment by samfundev** [comment]
Can you confirm that you can login to Twitch using your phone's browser?

**Comment by KandiPanda** [comment]
Okay, first I tried using the built-in browser (the one simply named "Browser" with a globe icon). It says that twitch.tv and m.twitch.tv are "unavailable". (And with a very similar look to the error page as trying to log in in Twire, complete with the little green robot peeking down from the top.)

Next I tried using the browser I actually use most of the time, one called Dolphin. It can load both those sites, but twitch.tv almost completely fails to render, and with m.twitch.tv (which loads and renders everything but the actual stream pages just fine) I can't find a login button anywhere (and typing in m.twitch.tv/login just redirects to www.twitch.tv/login for whatever reason, which then completely fails to render at all, but doesn't give an "unavailable" error either).

So I can't confirm that I can log in in the browser, but I *can* confirm that I can access the site (and load the login page, even if it doesn't render at all) in at least one browser, so it's not being *blocked* at least.

**Comment by KandiPanda** [comment]
Hm, I may have found a clue what my problem may be...

Was looking around, read through https://github.com/twireapp/Twire/pull/107 and saw this:

> Known issues
>-  ...
>- **Logging into a Twitch account may not be possible if the WebView does not support TLS 1.2; also happened on the Android 4.2.2 device.**

I can't say for certain if that's what's causing my issue or not, but it sounds like that _could_ possibly be the case?

**Comment by samfundev** [comment]
@KandiPanda That definitely sounds like it could be the issue. Unfortunately, I don't know of anything I can do to fix something like that. If you find anything that Twire could do to fix that issue, I could see if I could implement it. For now, I'm going to close this issue because I don't think there is anything I can do about this.

**Comment by KandiPanda** [comment]
Did some searching around, and found what appears to be a couple of variations on a solution for enabling TLS 1.2 in WebView.

https://github.com/Leeewy/TLSWebView

https://github.com/square/okhttp/issues/2372#issuecomment-244807676

...both of which seem to involve OkHttp, which coincidentally appears to already be present in Twire anyways from what I can tell.

**Comment by KandiPanda** [comment]
@samfundev Oh, just occurred to me that you might not've seen my comment above this one if you're not subscribed to this issue for whatever reason.

**Comment by samfundev** [comment]
Nope, I get notifications about everything in Twire. I'll look into seeing if I can implement one of those solutions into Twire.

**Comment by LukasThyWalls** [comment]
Also, there is another solution as does Xtra Twitch App, which is not clean and nice, but it works.

When you login, it give you the option to open the login page in a browser that works. When you log in, it appears a error page (which begans in "localhost...") and you can copy that url, go to the login again in Xtra and give you the option to paste that url. And once you did that, it loads normally, and you are log in. I tested the other day in my old 4.1.1 tablet and it worked.

**Comment by acjones8** [comment]
Just in case it helps anyone attempting to work on this bug: since this report was opened, Xtra went open source, and all of the code is available [here](https://github.com/AndreyAsadchy/Xtra). Both Xtra and Twire are GPL 3 I believe, so there shouldn't be any licensing issues with looking at or copying code between the projects.
